backend/src/main/java/hanglog/trip/service/DayLogService.java

```diff
@@ -50,4 +60,36 @@ private void validateAlreadyDeleted(final DayLog dayLog) {
             throw new BadRequestException(ALREADY_DELETED_DATE);
         }
     }
+
+    public void updateOrdinalOfItems(final Long dayLogId,
+                                     final ItemsOrdinalUpdateRequest itemsOrdinalUpdateRequest) {
+        final DayLog dayLog = dayLogRepository.findById(dayLogId)
+                .orElseThrow(() -> new BadRequestException(NOT_FOUND_DAY_LOG_ID));
+        final List<Item> items = dayLog.getItems();
+
+        final List<Long> orderedItemIds = itemsOrdinalUpdateRequest.getItemIds();
+        validateOrderedItemIds(items, orderedItemIds);
+        changeOrdinalOfItemsByOrderedItemIds(orderedItemIds);
+    }
+
+    private void validateOrderedItemIds(final List<Item> items, final List<Long> orderedItemIds) {
+        final Set<Long> itemIds = items.stream()
+                .map(Item::getId)
+                .collect(Collectors.toSet());
+        final Set<Long> orderedItemIdsSet = new HashSet<>(orderedItemIds);
```

Comment:
이 부분이 검증을 하는 것이 `orderedItemIds`가 가진 요소들이 정확히 itemIds가 가지고 있는지 판별하는 것이라고 생각하는데 set으로 받아주는 것 보다 sort해서 정확히 같은지 비교하는 방법은 어떤가요?
```suggestion
               final List<Long> itemIds = items.stream()
                .map(item->item.getId())
                .sorted()
                .collect(Collectors.toList());
        final List<Long> sortedOrderItemIds = orderedItemIds.stream().sorted().collect(Collectors.toList());

        if(!itemIds.equals(sortedOrderItemIds)){
            throw new BadRequestException(INVALID_ORDERED_ITEM_IDS);
        }
```